ios/RIBs/Classes/DI/Component.swift

```diff
@@ -47,14 +47,6 @@ open class Component<DependencyType>: Dependency {
             lock.unlock()
         }
 
-        #if swift(>=5.0)
```

Comment:
I'm assuming this was added with the idea that SR-8704 would be resolved with the release of Swift 5. It's unclear to me how that particular Swift bug was addressed (it is now closed), but regardless, the behavior you addressed in the #else block is still required. Thus, I have removed the #if swift(>=5.0) branch.

Reply:
@dzzh this is something you had the chance to catch in #274 and address with that very helpful test. I am eager to hear anything you have to say and suggest any further improvements you think are appropriate here.

Reply:
I think this is fine for now. I'd like to remove DI completely from this repo, since this is independent of RIBs.